src/go/rpk/pkg/cli/cluster/quotas/alter.go

```diff
@@ -0,0 +1,194 @@
+// by the Apache License, Version 2.0
+
+package quotas
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"go.uber.org/zap"
+)
+
+type createResponse struct {
+	Entity []entityData `json:"entity" yaml:"entity"`
+	Status string       `json:"status" yaml:"status"`
+
+	entityStr string
+}
+
+func alterCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		dry      bool
+		names    []string
+		defaults []string
+		adds     []string
+		deletes  []string
+	)
+	cmd := &cobra.Command{
+		Use:   "alter",
+		Args:  cobra.NoArgs,
+		Short: "Add or delete a client quota",
+		Long: `Add or delete a client quota
+
+This command allows you to add or delete a client quota.
+
+A client quota consists of an entity (to whom the quota is applied) and a quota 
+type (what is being applied).
+
+There are two entity types supported by Redpanda: client ID and client ID 
+prefix.
+
+Assigning quotas to default entity types is possible using the '--default' flag.
+
+You can perform a dry run using the '--dry' flag.
+`,
+		Example: `
+Add quota (consumer_byte_rate) to client ID 'foo':
+  rpk cluster quotas alter --add consumer_byte_rate=200000 \
+    --name client-id=foo
+
+Add quota (consumer_byte_rate) to client ID starting with 'bar-':
+  rpk cluster quotas alter --add consumer_byte_rate=200000 \
+    --name client-id-prefix=bar-
+
+Add quota (producer_byte_rate) to default client ID:
+  rpk cluster quotas alter --add producer_byte_rate=180000 \
+    --default client-id=foo
+
+Remove quota (producer_byte_rate) from client ID 'foo':
+  rpk cluster quotas alter --delete producer_byte_rate \
+    --name client-id=foo
+`,
+		Run: func(cmd *cobra.Command, _ []string) {
+			f := p.Formatter
+			if h, ok := f.Help(createResponse{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			var (
+				entity  []kadm.ClientQuotaEntityComponent
+				nameMap = make(map[string]bool)
+			)
+			for _, name := range names {
+				split := strings.SplitN(name, "=", 2)
+				if len(split) != 2 {
+					out.Die("name %q missing value", split[0])
+				}
+				k, v := split[0], split[1]
+				k = strings.ToLower(k)
+				if !validTypes[k] {
+					out.Die("name type %q is invalid (allowed: client-id, client-id-prefix)", split[0])
+				}
+				nameMap[k] = true
+				entity = append(entity, kadm.ClientQuotaEntityComponent{
+					Type: k,
+					Name: &v,
+				})
+			}
+			for _, def := range defaults {
+				if !validTypes[def] {
+					out.Die("default type %q is invalid (allowed: client-id, client-id-prefix)", def)
+				}
```

Comment:
The error message says that `--default client-id-prefix` is allowed even though it's not supported. I think there should probably be a separate `validTypes` for `any` and `default`. Alternatively, we could handle the validation in redpanda and rpk will just show the error but at least this error message should be changed.

Reply:
I added your suggestion in a new commit, thanks! https://github.com/redpanda-data/redpanda/pull/18711/commits/0ac43ee9a4a6c22c7614a1e823dce9128e0652b5